test/infrastructure/docker/internal/controllers/dockermachine_controller.go

```diff
@@ -304,17 +307,21 @@ func (r *DockerMachineReconciler) reconcileNormal(ctx context.Context, cluster *
 		}
 	}
 
-	// if the machine is a control plane update the load balancer configuration
-	// we should only do this once, as reconfiguration more or less ensures
-	// node ref setting fails
-	if util.IsControlPlaneMachine(machine) && !dockerMachine.Status.LoadBalancerConfigured {
-		unsafeLoadBalancerConfigTemplate, err := r.getUnsafeLoadBalancerConfigTemplate(ctx, dockerCluster)
-		if err != nil {
-			return ctrl.Result{}, errors.Wrap(err, "failed to retrieve HAProxy configuration from CustomHAProxyConfigTemplateRef")
-		}
-		if err := externalLoadBalancer.UpdateConfiguration(ctx, unsafeLoadBalancerConfigTemplate); err != nil {
-			return ctrl.Result{}, errors.Wrap(err, "failed to update DockerCluster.loadbalancer configuration")
+	if !dockerCluster.Spec.LoadBalancer.Disable {
```

Comment:
Should not be needed. control plane machines without the loadbalancer should not be supported IMHO.